### PR TITLE
Make pytorch model supporting model load-time param 'device'

### PR DESCRIPTION
**Pull request by WeichenXu123**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/WeichenXu123/mlflow/pull/10755?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10755/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10755
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?
Make pytorch model supporting model load-time param 'device'

Current master code setting model device in `_PyTorchWrapper.predict`, which is not a good way.

Manual test:
```
import torch
import mlflow

m1 = torch.nn.Linear(3, 5)
model_meta = mlflow.pytorch.log_model(m1, artifact_path="model")

model = mlflow.pyfunc.load_model(model_meta.model_uri, model_config={'device': 'cuda'})
```

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by WeichenXu123** [comment]
@mlflow-automation autoformat

**Comment by WeichenXu123** [comment]
@mlflow-automation autoformat

File: mlflow/pytorch/__init__.py [hunk 1]
```diff
@@ -685,13 +691,28 @@ def load_model(model_uri, dst_path=None, **kwargs):
...
+def _load_pyfunc(path, model_config):
     """
     Load PyFunc implementation. Called by ``pyfunc.load_model``.
 
     :param path: Local filesystem path to the MLflow Model with the ``pytorch`` flavor.
     """
-    return _PyTorchWrapper(_load_model(path, **kwargs))
+    import torch
+
+    device = model_config.get("device", None)
```
Comment (dbczumar) [inline mlflow/pytorch/__init__.py:702]: Can `model_config` be `None`? I noticed we check for it in the transformers flavor

File: mlflow/pytorch/__init__.py [hunk 2]
```diff
@@ -715,18 +737,13 @@ def predict(self, data, params: Optional[Dict[str, Any]] = None):
...
-        if device is None:
-            if MLFLOW_DEFAULT_PREDICTION_DEVICE.get():
-                device = MLFLOW_DEFAULT_PREDICTION_DEVICE.get()
-            elif torch.cuda.is_available():
-                device = _TORCH_DEFAULT_GPU_DEVICE_NAME
-            else:
-                device = _TORCH_CPU_DEVICE_NAME
+        if params and "device" in params:
+            raise ValueError(
+                "device' param is no longer inference param but becomes load-time param, "
```
Comment (dbczumar) [inline mlflow/pytorch/__init__.py:742]:
```suggestion
                "device' can no longer be specified as an inference parameter. It must be specified at load time. "
```

File: mlflow/pytorch/__init__.py [hunk 3]
```diff
@@ -715,18 +737,13 @@ def predict(self, data, params: Optional[Dict[str, Any]] = None):
...
+                "please use command like "
```
Comment (dbczumar) [inline mlflow/pytorch/__init__.py:743]:
```suggestion
                "Please specify the device at load time, for example: "
```

**Review by dbczumar — APPROVED** [review]
LGTM with small comments - thanks @WeichenXu123 !

**Pull request merged**